contracts/rewards/src/contract/execute.rs

```diff
@@ -149,6 +185,67 @@ where
 
         Ok(())
     }
+
+    fn get_pool_balance(&self, contract_addr: Addr) -> Result<Uint256, ContractError> {
+        match self.store.load_rewards_pool(contract_addr.clone())? {
+            Some(pool) => Ok(pool.balance),
+            None => Ok(Uint256::zero()),
+        }
+    }
+
+    fn update_pool_balance(
+        &mut self,
+        contract: Addr,
+        old_balance: Uint256,
+        workers_to_reward: &nonempty::Vec<Addr>,
+        rewards_per_worker: Uint256,
+    ) -> Result<(), ContractError> {
+        self.store.save_rewards_pool(&RewardsPool {
+            contract,
+            balance: old_balance
+                - (rewards_per_worker * Uint256::from(workers_to_reward.len() as u128)),
+        })
+    }
+}
+
+fn get_workers_to_reward(
+    tally: &EpochTally,
+) -> Result<nonempty::Vec<Addr>, axelar_wasm_std::nonempty::Error> {
```

Comment:
you're right that it's an ok scenario for no workers to be rewarded. I just found the logic a bit simpler when this function returns a `nonempty::Vec`, since then we don't have to check for zero length when processing the rewards.

Reply:
We shouldn't have to check that either way.